From 29c2db697e9e61c153ff8ed5d7869ff3bea06ffb Mon Sep 17 00:00:00 2001
From: Richard Kuo <rkuo@codeaurora.org>
Date: Tue, 20 Aug 2013 15:53:10 -0500
Subject: [PATCH 3/8] v5 platform fence setting

---
 linux/loadlinux.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/linux/loadlinux.c b/linux/loadlinux.c
index 8f9abcc..b6d3948 100644
--- a/linux/loadlinux.c
+++ b/linux/loadlinux.c
@@ -81,8 +81,14 @@ unsigned long vm_setup(char num_cpus, short num_ints, u32_t trans, unsigned long
 	}
 
 	if (pt == H2K_ASID_TRANS_TYPE_OFFSET) {
+#if ARCHV == 4
 		/* Memory map:  0 ... Linux ... frame buffer ... H2 ... boot VM ... ucos */
 		ret = h2_config_vmblock_init(vm, SET_FENCES, 0x0, FRAME_BUFFER);
+#endif
+#if ARCHV == 5
+		/* Memory map:  0 ... H2 ... Linux ... ?  */
+		ret = h2_config_vmblock_init(vm, SET_FENCES, 0x0, 0xfe000000);
+#endif
 		if (ret != vm) {
 			PRINTF("ret %08x\n", (unsigned int)ret);
 			FAIL("SET_FENCES");
-- 
1.7.8.3

