From 08e5b9b165bb9ee6a0df9da214c598035e92ecff Mon Sep 17 00:00:00 2001
From: Richard Kuo <rkuo@codeaurora.org>
Date: Tue, 20 Aug 2013 16:35:39 -0500
Subject: [PATCH 4/8] fix from Bryan for vmwait/interrupt something or rather

---
 kernel/vm/vmwork/vmwork.ref.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/kernel/vm/vmwork/vmwork.ref.c b/kernel/vm/vmwork/vmwork.ref.c
index aa045f6..27e8b7b 100644
--- a/kernel/vm/vmwork/vmwork.ref.c
+++ b/kernel/vm/vmwork/vmwork.ref.c
@@ -16,9 +16,11 @@ s32_t H2K_vm_do_work(H2K_thread_context *me)
 	
 	if (!(me->vmstatus & H2K_VMSTATUS_VMWORK)) return -1;
 
-	/* Can clear vmwork unconditionally here.  If interrupts are disabled or
-	 masked we'll check again on enable/unmask*/
-	H2K_atomic_clrbit(&me->atomic_status_word,H2K_VMSTATUS_VMWORK_BIT);
+       /* Need to leave vmwork bit set if interrupts are disabled in case we go to
+		vmwait before re-enabling */
+       if (me->vmstatus & H2K_VMSTATUS_IE) {
+	       H2K_atomic_clrbit(&me->atomic_status_word,H2K_VMSTATUS_VMWORK_BIT);
+       }
 
 	return H2K_vm_check_interrupts(me);
 }
-- 
1.7.8.3

