From fdab0dc1555dbdd8695f52f9eb7fdcfb7b58457f Mon Sep 17 00:00:00 2001
From: Richard Kuo <rkuo@codeaurora.org>
Date: Tue, 1 Oct 2013 20:00:05 -0500
Subject: [PATCH] switch to ctlbw for v5

---
 kernel/mem/tlbfill/tlbfill.ref.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/kernel/mem/tlbfill/tlbfill.ref.c b/kernel/mem/tlbfill/tlbfill.ref.c
index 07b88fa..6c88990 100644
--- a/kernel/mem/tlbfill/tlbfill.ref.c
+++ b/kernel/mem/tlbfill/tlbfill.ref.c
@@ -24,6 +24,7 @@ static inline u32_t H2K_mem_tlb_v3_user_check(H2K_thread_context *me)
 static inline void H2K_mem_tlb_insert(H2K_mem_tlbfmt_t entry, H2K_thread_context *me)
 {
 	u64_t rawentry;
+	u32_t result;
 	u32_t index = H2K_gp->tlb_index;
 	if ((index+1) < MAX_TLB_ENTRIES) {
 		H2K_gp->tlb_index = index+1;
@@ -46,14 +47,14 @@ static inline void H2K_mem_tlb_insert(H2K_mem_tlbfmt_t entry, H2K_thread_context
 	rawentry = entry.raw;
 	asm volatile
 		(
-		 " tlbw(%0,%1)\n"
 #if ARCHV == 5
-		 " nop\n"
-		 " nop\n"
-		 " nop\n"
+		 " %0 = ctlbw(%1,%2)\n"
+#else
+		 " tlbw(%1,%2)\n"
 #endif
 		 " isync\n"
-		 : : "r"(rawentry),"r"(index));
+		 : "=&r" (result)
+		 : "r"(rawentry),"r"(index));
 #endif
 }
 
-- 
1.7.12.1

