From 6cd0cd3dc0141b72e63a0a156969aae51de2c037 Mon Sep 17 00:00:00 2001
From: Richard Kuo <rkuo@codeaurora.org>
Date: Sat, 5 Oct 2013 14:50:08 -0500
Subject: [PATCH 1/2] 8974 MSS interrupt configs

---
 kernel/data/intconfig/intconfig.ref.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/kernel/data/intconfig/intconfig.ref.c b/kernel/data/intconfig/intconfig.ref.c
index 6028c51..8796e70 100644
--- a/kernel/data/intconfig/intconfig.ref.c
+++ b/kernel/data/intconfig/intconfig.ref.c
@@ -121,6 +121,18 @@ static void H2K_intconfig_l2_init()
 		#define TLMM_WORD	(TLMM_IRQ/32)
 #endif
 #if ARCHV >= 5
+
+#ifdef MODEM
+		#define SPI_IRQ		245
+		#define SPI_WORD	(SPI_IRQ/32)
+		#define TLMM_IRQ	239
+		#define TLMM_WORD	(TLMM_IRQ/32)
+
+		#define SPMI_ARB_PERIPH_IRQ	75
+		#define SPMI_ARB_PERIPH_WORD	(SPMI_ARB_PERIPH_IRQ/32)
+		#define SPMI_ARB_EE_IRQ		76
+		#define SPMI_ARB_EE_WORD	(SPMI_ARB_EE_IRQ/32)
+#else
 		/*  8974 SPI ("blsp1_qup1") and "summary_irq_sensors"  */
 		#define SPI_IRQ		64
 		#define SPI_WORD	(SPI_IRQ/32)
@@ -131,6 +143,7 @@ static void H2K_intconfig_l2_init()
 		#define SPMI_ARB_PERIPH_WORD	(SPMI_ARB_PERIPH_IRQ/32)
 		#define SPMI_ARB_EE_IRQ		49
 		#define SPMI_ARB_EE_WORD	(SPMI_ARB_EE_IRQ/32)
+#endif  //  !MODEM
 #endif
 		/*  is there a "clear bit" around here?  */
 		/*  There's probably a better way to do this.  */
-- 
1.7.12.1

